src/node_jsrtapi.cc

```diff
@@ -1004,67 +1047,157 @@ napi_status napi_coerce_to_string(napi_env e, napi_value v, napi_value* result)
 }
 
 napi_status napi_wrap(napi_env e, napi_value jsObject, void* nativeObj,
-  napi_destruct destructor, napi_weakref* handle) {
-  new ObjectWrapWrapper(jsObject, nativeObj, destructor);
+    napi_finalize finalize_cb, napi_ref* result) {
+  JsValueRef value = reinterpret_cast<JsValueRef>(jsObject);
 
-  if (handle != nullptr)
-  {
-    CHECK_NAPI(napi_create_weakref(e, jsObject, handle));
+  jsrtimpl::ExternalData* externalData = new jsrtimpl::ExternalData(nativeObj, finalize_cb);
+  CHECK_JSRT(JsSetExternalData(value, externalData));
+
+  if (result != nullptr) {
+    napi_create_reference(e, jsObject, 0, result);
   }
+
   return napi_ok;
 }
 
 napi_status napi_unwrap(napi_env e, napi_value jsObject, void** result) {
-  CHECK_ARG(result);
-  *result = ObjectWrapWrapper::Unwrap(jsObject);
-  return napi_ok;
+  return napi_get_value_external(e, jsObject, result);
 }
 
-napi_status napi_create_persistent(napi_env e, napi_value v, napi_persistent* result) {
+napi_status napi_create_external(napi_env e, void* data, napi_finalize finalize_cb,
```

Comment:
For Chakra it would be 'object'. For V8... I actually don't know.

They might be different, but it shouldn't matter because JavaScript code cannot interact with these kinds of objects in any meaningful way.